extensions/widgets/switchbutton/switchbutton.lcb

```diff
@@ -17,199 +17,709 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+	put color [175/255,175/255,175/255] into mAndroidLightThumbDisabled
+	put color [0,0,0,0.26] into mAndroidLightTrackOff
+	put color [0,0,0,0.12] into mAndroidLightTrackDisabled
+
+	put color [175/255,175/255,175/255] into mAndroidDarkThumbOff
+	put color [51/255,51/255,51/255] into mAndroidDarkThumbDisabled
+	put color [1,1,1,0.3] into mAndroidDarkTrackOff
+	put color [1,1,1,0.1] into mAndroidDarkTrackDisabled
+
+end handler
+
+private handler updateVariables() returns nothing
+	put my width into mWidth
+	put my height into mHeight
+	put mWidth/9 into mRadius
+end handler
+
+private handler clickableRect() returns Rectangle
+	if mWidgetTheme is "iOS" then
+		return rectangle [mWidth/2 - mHeight/2, mHeight/4, mWidth/2 + mHeight/2, 3*mHeight/4]
+	else if mWidgetTheme contains "Android" then
+		return rectangle [1.5*mRadius, mHeight/2 - mRadius, 6*mRadius, mHeight/2 + mRadius]
 	end if
 end handler
 
-public handler OnMouseUp()
-	if the click position is within getRect("bounding box") then
-		setChecked(not(mIsChecked))
-		put "not animating" into mState
+public handler OnMouseMove() returns nothing
+	if mIsPressed then
+		variable tXMousePos as Real
+		variable tChange as Real
+		variable tMovedRight as Boolean
+
+		put true into mMouseHasMoved
+
+		put the x of the mouse position into tXMousePos
+		put tXMousePos - mXClick into tChange
+		if tChange > 0 then
+			put true into tMovedRight
+		else
+			put false into tMovedRight
+		end if
+
+		if tMovedRight then
+			put true into mSwitchIsOn
+		else
+			put false into mSwitchIsOn
+		end if
+
 		redraw all
 	end if
+end handler
 
-	post "mouseUp" with [mIsChecked]
+public handler OnMouseDown() returns nothing
+	if the click position is within clickableRect() then
+		put the x of the click position into mXClick
+		put true into mIsPressed
+		redraw all
+	end if
 end handler
 
-private handler getPath(in pShape as String) returns Path
-	if pShape is "box" then
-		return rounded rectangle path of rectangle [(my width)/2 - (my height)/2, (my height)/4, (my width)/2 + (my height)/2, 3*(my height)/4] with radius (my height)/4
-	else if pShape is "left circle" then
-		return circle path centered at point [(my width)/2 - (my height)/4, (my height)/2] with radius (my height)/4
-	else if pShape is "right circle" then
-		return circle path centered at point [(my width)/2 + (my height)/4, (my height)/2] with radius (my height)/4
-	else if pShape is "ellipse" then
-		return ellipse path centered at point [(my width)/2, (my height)/2] with radii [(my height)/8, (my height)/4]
-	else if pShape is "left rect" then
-		return rectangle path of rectangle [(my width)/2 - (my height)/4, (my height)/4, (my width)/2, 3*(my height)/4]
-	else if pShape is "right rect" then
-		return rectangle path of rectangle [(my width)/2, (my height)/4, (my width)/2 + (my height)/4, 3*(my height)/4]
+public handler OnMouseUp() returns nothing
+	if mMouseHasMoved then
+		post "mouseUp" with [mSwitchIsOn]
```

Comment:
I think it should post "switchChanged" or something along those lines.

Also, messages that a widget sends need to be documented at the top of the file (see the `navigate` message at the top here: https://github.com/livecode/livecode/blob/develop/extensions/widgets/navbar/navbar.lcb)